### PR TITLE
Add support for OkHttp 4.4+

### LINKED ISSUE #1609: Support OkHttp3 clients >= 4.4.0
We currently support [OkHttp clients up to 4.3](https://github.com/elastic/apm-agent-java/pull/1608), but versions >= 4.4.0 do not seem to be supported yet - tests fail and no method is matched by instrumentation.

### PR DESCRIPTION
**Pull request by felixbarny**
## What does this PR do?

closes #1609

## Checklist

- [x] This is an enhancement of existing features, or a new feature in existing plugins
  - [x] I have updated [CHANGELOG.asciidoc](https://github.com/elastic/apm-agent-java/blob/master/CHANGELOG.asciidoc)
  - [x] I have added tests that prove my fix is effective or that my feature works
  - [ ] ~Added an API method or config option? Document in which version this will be introduced~
  - [x] I have made corresponding changes to the documentation
- [ ] This is a bugfix
  - [ ] I have updated [CHANGELOG.asciidoc](https://github.com/elastic/apm-agent-java/blob/master/CHANGELOG.asciidoc)
  - [ ] I have added tests that would fail without this fix
- [ ] This is a new plugin
  - [ ] I have updated [CHANGELOG.asciidoc](https://github.com/elastic/apm-agent-java/blob/master/CHANGELOG.asciidoc)
  - [ ] My code follows the [style guidelines of this project](https://github.com/elastic/apm-agent-java/blob/master/CONTRIBUTING.md#java-language-formatting-guidelines)
  - [ ] I have made corresponding changes to the documentation
  - [ ] I have added tests that prove my fix is effective or that my feature works
  - [ ] New and existing [**unit** tests](https://github.com/elastic/apm-agent-java/blob/master/CONTRIBUTING.md#testing) pass locally with my changes
  - [ ] I have updated [supported-technologies.asciidoc](https://github.com/elastic/apm-agent-java/blob/master/docs/supported-technologies.asciidoc)
  - [ ] Added an API method or config option? Document in which version this will be introduced
  - [ ] Added an instrumentation plugin? Describe how you made sure that old, non-supported versions are not instrumented by accident.
- [ ] This is something else
  - [ ] I have updated [CHANGELOG.asciidoc](https://github.com/elastic/apm-agent-java/blob/master/CHANGELOG.asciidoc)

### CODE REVIEW

File: CHANGELOG.asciidoc [hunk 1]
```diff
@@ -35,6 +35,7 @@ to assist with such efforts.
 * Update existing Hibernate Search 6 instrumentation to the final relase
 * The <<config-use-path-as-transaction-name, `use_path_as_transaction_name`>> option is now dynamic
 * Flushing internal and micrometer metrics before the agent shuts down - {pull}1658[#1658]
+* Support for OkHttp 4.4+
```
Comment (SylvainJuge) [inline CHANGELOG.asciidoc:38]: [minor] missing PR link

File: apm-agent-plugins/apm-okhttp-test/src/test/java/co/elastic/apm/agent/okhttp/OkHttp3ClientVersionIT.java [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+import co.elastic.apm.agent.TestClassWithDependencyRunner;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+
+import java.util.Arrays;
+import java.util.List;
+
+@RunWith(Parameterized.class)
+public class OkHttp3ClientVersionIT {
```
Comment (SylvainJuge) [inline apm-agent-plugins/apm-okhttp-test/src/test/java/co/elastic/apm/agent/okhttp/OkHttp3ClientVersionIT.java:12]: [minor] maybe remove the `3` in the class name as we only test version 4.x

File: apm-agent-plugins/apm-okhttp-test/src/test/java/co/elastic/apm/agent/okhttp/OkHttp3ClientVersionIT.java [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+        this.runner = new TestClassWithDependencyRunner(dependencies,
+            OkHttp3ClientInstrumentationTest.class);
+    }
+
+    @Parameterized.Parameters(name = "{0}")
+    public static Iterable<Object[]> data() {
+        return Arrays.asList(new Object[][]{
+            // ivy can't automatically resolve the dependencies, so we'll have to manually type them in
+            // check https://mvnrepository.com/artifact/com.squareup.okhttp3/okhttp for the respective versions of kotlin and okio
+            {List.of("com.squareup.okhttp3:okhttp:4.3.0", "org.jetbrains.kotlin:kotlin-stdlib:1.3.61", "com.squareup.okio:okio:2.4.1")},
```
Comment (SylvainJuge) [inline apm-agent-plugins/apm-okhttp-test/src/test/java/co/elastic/apm/agent/okhttp/OkHttp3ClientVersionIT.java:25]: why not have a 1.x and 2.x versions in this list also ? I know that earlier version might not depend on Kotlin, thus we could expect them to be quite different code-wise.

**Review by SylvainJuge — APPROVED** [review]

File: apm-agent-plugins/apm-okhttp-test/src/test/java/co/elastic/apm/agent/okhttp/OkHttp3ClientVersionIT.java (hunk 2 above)
Comment (felixbarny) [reply apm-agent-plugins/apm-okhttp-test/src/test/java/co/elastic/apm/agent/okhttp/OkHttp3ClientVersionIT.java:12]: It's named that way as it tests the `com.squareup.okhttp3` module. I've added some versions from the 3.x version line.

File: apm-agent-plugins/apm-okhttp-test/src/test/java/co/elastic/apm/agent/okhttp/OkHttp3ClientVersionIT.java (hunk 3 above)
Comment (felixbarny) [reply apm-agent-plugins/apm-okhttp-test/src/test/java/co/elastic/apm/agent/okhttp/OkHttp3ClientVersionIT.java:32]: These are tested in `OkHttpClientInstrumentationTest.java` (not OkHttp3)

**Pull request merged**